src/markov/markov_approx.jl

```diff
@@ -130,3 +130,87 @@ function _rouwenhorst(p::Real, q::Real, m::Real, Δ::Real, n::Integer)
         return linspace(m-Δ, m+Δ, n), θN
     end
 end
+
+
+# These are to help me order types other than vectors
+@inline _emcd_lt{T}(a::T, b::T) = isless(a, b)
+@inline _emcd_lt{T}(a::Vector{T}, b::Vector{T}) = Base.lt(Base.Order.Lexicographic, a, b)
+
+"""
+Accepts the simulation of a discrete state Markov chain and estimates
+the transition probabilities
+
+Let S = {s₁, s₂, ..., sₙ} with s₁ < s₂ < ... < sₙ be the discrete
+states of a Markov chain. Furthermore, let P be the corresponding
+stochastic transition matrix.
+
+Given a history of observations, {X} where xₜ ∈ S ∀ t, we would like
+to estimate the transition probabilities in P, pᵢⱼ. For xₜ=sᵢ and xₜ₋₁=sⱼ,
+let P(xₜ | xₜ₋₁) be the pᵢⱼ element of the stochastic matrix. The likelihood
+function is then given by
+
+    L({X}ₜ; P) = P(x_1) ∏_{t=2}^{T} P(xₜ | xₜ₋₁)
+
+The maximum likelihood estimate is then just given by the number of times
+a transition from sᵢ to sⱼ is observed divided by the number of times
+sᵢ was observed.
+
+Note: Because of the estimation procedure used, only states that are observed
+in the history appear in the estimated Markov chain... It can't divine whether
+there are unobserved states in the original Markov chain.
+
+For more info, refer to:
+
+- http://www.stat.cmu.edu/~cshalizi/462/lectures/06/markov-mle.pdf
+- https://stats.stackexchange.com/questions/47685/calculating-log-likelihood-for-given-mle-markov-chains
+
+##### Arguments
+
+- `X::Vector{T}` : Simulated history of Markov states
+
+##### Returns
+
+- `mc::MarkovChain{T}` : A Markov chain holding the state values and
+transition matrix
+
+"""
+function estimate_mc_discrete{T}(X::Vector{T}, states::Vector{T})
+    # Get length of simulation
+    capT = length(X)
+
+    # Make sure all of the passed in states appear in X... If not
+    # throw an error
+    if any(map(x->in(x, X), states) .== false)
+        error("One of the states does not appear in history X")
```

Comment:
Being picky now means that we spend less time maintaining the code later -- No apology is needed.

I've added a test for when it is given a state that doesn't show up in the history.